contract_compassion/wizard/end_contract_wizard.py

```diff
@@ -64,22 +64,25 @@ def end_contract(self):
         self.ensure_one()
         contract = self.contract_id
         child = contract.child_id
-        hold = child.hold_id
 
-        if not self.has_new_hold:
-            # child will be inactive after this choice
-            self.hold_expiration_date = fields.Datetime.now()
-            child.write({'active': False})
-
-        # in all case current hold became inactive
-        hold.write({'active': False})
+        if self.has_new_hold:

```

Comment:
I think you forgot the case where the sponsorship was not yet activated. In that scenario the child may still have a valid hold active. So in that case :
1. If user wants to keep the child on hold - do nothing
2. If user doesn't want to keep the hold - inactive it, remove it, and inactive the child (+ state = 'F')